src/app/test/test.component.ts

```diff
@@ -252,10 +253,8 @@ export class TestComponent implements OnInit {
       if (!path.startsWith('/')) {
         path = '/' + path;
       }
-      let map = { path: path, action: action };
-      this.httpService.updatePath(reportIds, this.currentView.storageName, map).subscribe(() => {
-        this.loadData(path);
-      });
+      let map: UpdatePathSettings = { path: path, action: action };
```

Comment:
If the name and the variable are the same, it is possible to do this.

```suggestion
      let map: UpdatePathSettings = { path, action };
```

Looks cool right! 

This is however harder to refactor (if you rename one of the variables), so it's fine like how you did it.